File: corpus/typescript/fixture_10.ts
type CartItem10 = { price: number; qty?: number };
export function calculateDiscount10(cartItems: CartItem10[], taxRate: number): number {
  const totalAmount = cartItems.reduce((sum, item) => sum + item.price * (item.qty ?? 1), 0);
  return totalAmount > 100 ? totalAmount * (1 + taxRate) : totalAmount;
}

export class PriceBook10 {
  lookupPrice(sku: string): number { return sku.length + 10; }
}

File: corpus/go/fixture_10.go
package fixture10

type PriceBook10 struct {
    Multiplier float64
}

func CalculateDiscount10(cartItems []float64, taxRate float64) float64 {
    totalAmount := 0.0
    for _, price := range cartItems {
        totalAmount += price
    }
    if totalAmount > 100.0 {
        return totalAmount * (1.0 + taxRate)
    }
    return totalAmount
}
